libraries/ESP8266WiFi/src/BearSSLHelpers.cpp

```diff
@@ -872,6 +872,22 @@ bool X509List::append(const uint8_t *derCert, size_t derLen) {
   return true;
 }
 
+ServerSessions::~ServerSessions() {
+  if (_isDynamic && _store != nullptr)
+    delete _store;
+}
+
+ServerSessions::ServerSessions(uint8_t *store, uint32_t size, bool isDynamic) :
```

Comment:
Can we change the `size` parameter here and elsewhere to be "count of sessions to store in cache" as opposed to "number of bytes"?  (i.e. hide the `* 100`).

If each session is 100 bytes, then it doesn't make sense to allow passing in 150 bytes (for example), since `size%100` will be just wasted

Reply:
I thought about it but I liked the idea of the user being aware of how much memory it would take so that the user doesn't decide to cache 1000 sessions. I do however agree that it makes more sense to do it your way.

A good in between could be to do it your way but have an extra method name something like `bufferSize` that returns the size of the buffer. What do you think?

Reply:
I appreciate your concern, especially in a general purpose system.

But, practically, I think a notice in the docs/source that "each session cache takes 100 bytes" is plenty of warning.   The number of entries seems like something that would be hardcoded in user apps, not a variable, anyway, so I'm not sure another call is required.  The buffer itself is opaque, so there's not much they could do with knowing it's size, anyway...

Reply:
Alright, I understand. I'll do the changes and update the documentation tomorrow.

Reply:
I've done it and I've updated the documentation and the examples